### PR TITLE
Updated Clarity Maps

### PR DESCRIPTION
**Pull request by setzeus**
## Description
Updates two structs in the sbtc-registry contract to stored data already passed through.
 
Closes: #800 

## Changes
The 'sbtc-registry' Clarity smart contract

## Testing Information
All unit tests effected are now updated.

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by djordon — COMMENTED** [review]
Can we also update the `get-withdrawal-request` read only function to flatten out the return tuple. Right now the `status` is a tuple that also contains another `status` field, but I think we want 
```clarity
(merge request (map-get? withdrawal-status id))
```
right?

File: contracts/contracts/sbtc-registry.clar [hunk 1]
```diff
@@ -41,14 +41,22 @@
 ;; If status is `none`, the request is pending.
 ;; Otherwise, the boolean value indicates whether
 ;; the deposit was accepted.
-(define-map withdrawal-status uint bool)
+(define-map withdrawal-status uint {
+  status: bool,
```
Comment (djordon) [inline contracts/contracts/sbtc-registry.clar:45]: We should rename this `is-accepted`

**Review by djordon — COMMENTED** [review]
You're right, the type is fine! But let's rename the field to `is-accepted`.

**Comment by hstove** [comment]
@setzeus I think Daniel meant that the `status` prop should be changed to `is-accepted`, but the `withdrawal-status` map should still be named that IMO.

**Comment by hstove** [comment]
I don't mean to shoehorn in complexity here, but because I've been thinking about costs lately I wanted to mention this.

This modification (increasing the size of these maps) greatly increases `read_length`, especially with the change to `withdrawal-status` (it's like a 100x+ increase in read_length).

It's still a `read_count` of 1, which is the most important metric to optimize, but IMO it's worth considering breaking the `sweep` fields into a separate map. These fields are never used during contract calls (writes), they're just used to store information for off-chain usage.

Even though read_count is all that matters really today, this might not be the case in the future, which is why I think this is worth considering.

**Comment by djordon** [comment]
> I don't mean to shoehorn in complexity here, but because I've been thinking about costs lately I wanted to mention this.
> 
> This modification (increasing the size of these maps) greatly increases `read_length`, especially with the change to `withdrawal-status` (it's like a 100x+ increase in read_length).
> 
> It's still a `read_count` of 1, which is the most important metric to optimize, but IMO it's worth considering breaking the `sweep` fields into a separate map. These fields are never used during contract calls (writes), they're just used to store information for off-chain usage.
> 
> Even though read_count is all that matters really today, this might not be the case in the future, which is why I think this is worth considering.

Great point, let's break the sweep fields into a separate map.

**Comment by setzeus** [comment]
Ah makes sense @hstove, I'll create a separate map.

**Review by hstove — APPROVED** [review]

**Comment by djordon** [comment]
@hstove does it make sense to break out things on the deposit side as well? I was thinking that we'd have a map with a boolean value for the status and another with the additional data, just like we do for withdrawals.

**Comment by hstove** [comment]
Yep I like it 👍🏼

**Review by djordon — COMMENTED** [review]
Awesome, this is the structure that I think we should go with. But with these latest changes I think we should rename the `completed-deposit-sweep` map to `completed-deposits`, have the existing `get-completed-deposit` return the completed-deposits map and add `get-deposit-status` and that returns the `deposit-status`. We'd also have to update our current usage of `get-completed-deposit` to use the `get-deposit-status` function.

**Review by djordon — APPROVED** [review]
Looks great! ✅ 🚢

**Pull request merged**